src/web_app/js/util.js

```diff
@@ -37,21 +37,40 @@ function queryStringToDictionary(queryString) {
 
 // Sends the URL request and returns a Promise as the result.
 function sendAsyncUrlRequest(method, url, body) {
+  return sendUrlRequest(method, url, true, body);
+}
+
+// If async is true, returns a Promise and executes the xhr request
+// async. If async is false, the xhr will be executed sync and a
+// resolved promise is returned.
+function sendUrlRequest(method, url, async, body) {
   return new Promise(function(resolve, reject) {
-    var xhr = new XMLHttpRequest();
-    xhr.onreadystatechange = function() {
-      if (xhr.readyState !== 4) {
-        return;
-      }
+    var reportResults = function() {
       if (xhr.status !== 200) {
         reject(
-            Error('Status=' + xhr.status + ', response=' + xhr.responseText));
+            Error('Status=' + xhr.status + ', response=' +
+                  xhr.responseText));
         return;
       }
       resolve(xhr.responseText);
     };
-    xhr.open(method, url, true);
+
+    var xhr;
+    xhr = new XMLHttpRequest();
+    if (async) {
+      xhr.onreadystatechange = function() {
```

Comment:
Interesting, I had thought it wasn't, but it appears that at least in Chrome and FF it is called even with async=false.